app/server/lib/Requests.ts

```diff
@@ -91,7 +91,7 @@ export class DocRequests {
       }
       const response = await fetch(urlObj.toString(), {
         headers: headers || {},
-        agent: proxyAgent(urlObj),
+        agent: proxyAgentForTrustedRequests(urlObj),
```

Comment:
Should be untrusted, as this module is used for the `REQUEST()` function.
Quoting the README file:
> GRIST_ENABLE_REQUEST_FUNCTION 	enables the REQUEST function. This function performs HTTP requests in a similar way to requests.request. This function presents a significant security risk, since it can let users call internal endpoints when Grist is available publicly. This function can also cause performance issues. Unset by default.

```suggestion
        agent: proxyAgentForUntrustedRequests(urlObj),
```

Reply:
I am lost. 😞 OIDCCOnfig tests fail because it is contradictory. Indeed, if GRIST_HTTPS_PROXY enables proxyAgentForUntrustedRequests and HTTPS_PROXY enables proxyAgentForTrustedRequests as ProxyAgent.tx does, then if requests uses proxyAgentForUntrustedRequests we must enable GRIST_HTTPS_PROXY with the same corporate proxy value, but then why two envs ?

Reply:
Why two environment variables: as suggested by Paul above, it is useful to be able to distinguish requests that are made to URLs determined by an administrator, e.g. requests to an OIDC provider — we call those "trusted"; from requests made to URLs that are given by Grist end users (i.e. Webhooks and requests made through the `REQUEST` function if you enable it), over which the administrator has no control — "untrusted" requests.

Having both `HTTPS_PROXY` for trusted requests and `HTTPS_PROXY_UNTRUSTED_URLS` (new name for `GRIST_HTTPS_PROXY` suggested by Paul) for untrusted requests can let an administrator apply more stringent controls on untrusted requests.

For example, in your case you may want to use a lenient outgoing proxy for `HTTPS_PROXY`, that allows any request to the internet (including those to your OIDC provider); and set up a stricter HTTP proxy pointed to by `HTTPS_PROXY_UNTRUSTED_URLS` for untrusted requests, for example one that only allows requests to a limited list of domains, and/or limits request verbs, sizes, etc.

Reply:
@tristanrobert Will you still make the changes suggested above? I find myself in agreement that `HTTPS_PROXY_UNTRUSTED_URLS` is a better name, as well as noting that `REQUEST` should use the proxy for untrusted URLs.

Reply:
I am working on it, and I agree that this would be a better name!